### PR TITLE
Implement Mentor Profile Enhancements and Matching Page Functionality

### PR DESCRIPTION
**Pull request by Domm6**
- Added a bio attribute for all mentors.
- Implemented a modal that opens when a mentor card is clicked.
- Integrated a company logo API based on the user's company.
- Fixed the form bio text box in the mentor profile modal.
- Created a route to fetch all mentor data.
- Correctly fetches the name of the user associated with a mentor.
- Added company logo API to the matching page.
- Ensured the matching modal displays mentor data correctly.
- Displayed mentor bio when hovering over mentor cards in the matching section.
- Removed all console logging for security purposes.
- Updated mentor profile design.
- Fixed text wrapping issues when hovering over mentor cards.

Images:
Screenshot 2024-06-28 at 11 12 09 AM
Screenshot 2024-06-28 at 11 12 22 AM
Screenshot 2024-06-28 at 11 12 36 AM
Screenshot 2024-06-28 at 11 13 15 AM

### CODE REVIEW

File: Frontend/src/components/mentor-match/MentorCard.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    </div>
+                    <div className='mc-role'>
+                        <p>{mentor.work_role}</p>
+                    </div>
+                    <div className='mc-company'>
+                        <p>{mentor.company}</p>
+                    </div>
+                </div>
+                <div className='mc-body-right'>
+                    <img src={`https://img.logo.dev/${mentor.company}.com?token=pk_DCOxK2D7TA68fkEDQQ2_fQ`} />
```
Comment (jzwang43) [inline Frontend/src/components/mentor-match/MentorCard.jsx]: good use of the API, we can also move this base URL into a const

**Review by jzwang43 — APPROVED** [review]
Overall looks good, left a couple nits and a question

**Comment by jzwang43** [comment]
there's one commit called "wq" which is a bit confusing maybe next time can avoid that one, but other commits are named very well

**Comment by jzwang43** [comment]
Do we know if the vite/deps need to be commited, if not we can remove those with gitignore

File: Frontend/src/components/MentorProfile.css [hunk 2]
```diff
@@ -59,6 +59,24 @@
...
+.mp-right p {
+    box-shadow: 0 2px 5px rgba(0,0,0,0.2);
+    border-radius: 8px;
+    padding: 10px;
+}
+
+.mp-right-company {
+    display: flex;
+    align-items: center;
+
```
Comment (sbrina813) [inline Frontend/src/components/MentorProfile.css]: nit: remove line

File: Frontend/src/components/MentorProfileModal.css [hunk 3]
```diff
@@ -69,4 +71,11 @@
 
 .skills-dropdown div {
     margin-bottom: 5px;
+}
+
```
Comment (sbrina813) [inline Frontend/src/components/MentorProfileModal.css:75]: nit: remove line

File: Frontend/src/components/MentorProfile.css (hunk 2 above)
Comment (Domm6) [reply Frontend/src/components/MentorProfile.css]: Resolved

**Pull request merged**